core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -84,27 +89,46 @@ public static StructLikeSet toEqualitySet(CloseableIterable<StructLike> eqDelete
   }
 
   public static <T extends StructLike> PositionDeleteIndex toPositionIndex(CharSequence dataLocation,
-                                                                           List<CloseableIterable<T>> deleteFiles) {
+                                                                           List<CloseableIterable<T>> deleteFiles,
+                                                                           DeleteCounter counter) {
     DataFileFilter<T> locationFilter = new DataFileFilter<>(dataLocation);
     List<CloseableIterable<Long>> positions = Lists.transform(deleteFiles, deletes ->
         CloseableIterable.transform(locationFilter.filter(deletes), row -> (Long) POSITION_ACCESSOR.get(row)));
-    return toPositionIndex(CloseableIterable.concat(positions));
+    return toPositionIndex(CloseableIterable.concat(positions), counter);
   }
 
-  public static PositionDeleteIndex toPositionIndex(CloseableIterable<Long> posDeletes) {
+  public static PositionDeleteIndex toPositionIndex(CloseableIterable<Long> posDeletes, DeleteCounter counter) {
     try (CloseableIterable<Long> deletes = posDeletes) {
       PositionDeleteIndex positionDeleteIndex = new BitmapPositionDeleteIndex();
       deletes.forEach(positionDeleteIndex::delete);
+      counter.increment(((BitmapPositionDeleteIndex) positionDeleteIndex).numberOfPositionsDeleted());
```

Comment:
we can the skip the cast here if we just keep the type on 102

Reply:
If we add `numberOfPositionsDeleted()` in the `PositionDeleteIndex` interface, then we don't need the cast either.